src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -241,26 +178,159 @@ std::string postQueryContext
   //
-  if ((qcrsP->errorCode.code == SccOk) || (qcrsP->errorCode.details == ""))
+  if (qcrsP->errorCode.code == SccNone)
   {
-    qcrsP->errorCode.details = std::string("Redirected to context provider ") + ip + ":" + portV + prefix;
+    qcrsP->errorCode.fill(SccOk);
   }
-  else if (qcrsP->errorCode.code == SccNone)
+
+  if ((qcrsP->contextElementResponseVector.size() == 1) && (qcrsP->contextElementResponseVector[0]->statusCode.code == SccContextElementNotFound))
   {
-    qcrsP->errorCode.fill(SccOk, std::string("Redirected to context provider ") + ip + ":" + portV + prefix);
+    qcrsP->errorCode.fill(SccContextElementNotFound);
   }
+}
+
+
 
-  if (qcrsP->contextElementResponseVector.size() > 0)
+/* ****************************************************************************
+*
+* postQueryContext -
+*/
+std::string postQueryContext
+(
+  ConnectionInfo*            ciP,
+  int                        components,
+  std::vector<std::string>&  compV,
+  ParseData*                 parseDataP
+)
+{
+  //
+  // Convops calling this routine may need the response in digital
+  // So, the digital response is passed back in parseDataP->qcrs.res
+  //
+  QueryContextResponse*       qcrsP = &parseDataP->qcrs.res;
+  std::string                 answer;
+  int                         noOfForwards = 0;
+  QueryContextRequestVector   requestV;
+  QueryContextResponseVector  responseV;
+
+  qcrsP->errorCode.fill(SccOk);
+  ciP->httpStatusCode = mongoQueryContext(&parseDataP->qcr.res, qcrsP, ciP->tenant, ciP->servicePathV, ciP->uriParam);
+  qcrsP->present("", "postQueryContext");
+
+  for (unsigned int ix = 0 ; ix < qcrsP->contextElementResponseVector.size(); ++ix)
+  {
+    ContextElementResponse* cerP       = qcrsP->contextElementResponseVector[ix]->clone();
+
+    for (unsigned int aIx = 0 ; aIx < cerP->contextElement.contextAttributeVector.size(); ++aIx)
+    {
+      EntityId*            eP       = &cerP->contextElement.entityId;
+      ContextAttribute*    aP       = cerP->contextElement.contextAttributeVector[aIx];
+      QueryContextRequest* requestP = requestV.lookup(aP->providingApplication, eP);   // No need to lookup if providingApplication == "" ...
+      
+      //
+      // An empty providingApplication means the attribute is local
+      // In such a case, the response is already in our hand, we just need to forward it to responseV
+      //
+      if (aP->providingApplication == "")
+      {
+        if (aP->found == false)
+        {
+          continue;  // Non-found pairs of entity/attribute are thrown away
+        }
+
+        QueryContextResponse* qP = new QueryContextResponse(eP, aP);
+        responseV.push_back(qP);
+      }
+      else if (requestP == NULL)
+      {
+        requestP = new QueryContextRequest(aP->providingApplication, eP, aP->name);
+        requestV.push_back(requestP);
+        ++noOfForwards;
+      }
+      else
+      {
+        requestP->attributeList.push_back_if_absent(aP->name);
```

Comment:
I understand that with this strategy, if you do the following query E-[A, B, C, D] resulting in:

E-A -> CPr1
E-B -> CPr1
E-C -> CPr2
E-D -> CPr2

At the end 2 forward messages are sent (one for CPr1 with E1-[A,B] and other for E-[C,D]) and not 4 messages (2 to each CPr, containing each message 2 attributes).

Is this correct?